src/librustc_metadata/locator.rs

```diff
@@ -845,6 +860,15 @@ fn get_metadata_section_imp(target: &Target,
                 Ok(blob)
             }
         };
+    } else if flavor == CrateFlavor::Rmeta {
+        let mut file = File::open(filename).map_err(|_|
+            format!("could not open file: '{}'", filename.display()))?;
```

Comment:
Could this include the reason it failed to open as well? (e.g. the argument to `map_err`)